client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx

```diff
@@ -463,25 +419,32 @@ export default function WPCheckout( {
 							reduxDispatch,
 							translate,
 							true
-						).then( ( response ) => {
-							if ( response ) {
-								// When the contact details change, update the cached contact details on
-								// the server. This can fail if validation fails but we will silently
-								// ignore failures here because the validation call will handle them better
-								// than this will.
-								reduxDispatch(
-									saveContactDetailsCache( prepareDomainContactValidationRequest( contactInfo ) )
-								);
-
-								reduxDispatch(
-									recordTracksEvent( 'calypso_checkout_composite_step_complete', {
-										step: 1,
-										step_name: 'contact-form',
-									} )
-								);
-							}
-							return response;
-						} );
+						);
+						if ( validationResponse ) {
+							// When the contact details change, update the cart's tax location to match.
+							await updateCartContactDetailsForCheckout(
```

Comment:
Previously we did not wait for this part (the cart update) to finish before proceeding to the next step, on the theory that the form should be disabled anyway when a cart update is in-progress, but being more explicit feels safer to me and should avoid possible race conditions.